src/char/sorceress/sorceress.py

```diff
@@ -32,6 +33,40 @@ def pick_up_item(self, pos: Tuple[float, float], item_name: str = None, prev_cas
             return cast_start
         else:
             return super().pick_up_item(pos, item_name, prev_cast_start)
+            
+    def select_by_template(
+        self,
+        template_type:  Union[str, List[str]],
+        success_func: Callable = None,
+        time_out: float = 8,
+        threshold: float = 0.68,
+        telekinesis = False
+    ) -> bool:
+        if type(template_type) == list and "A5_STASH" in template_type:
+            # sometimes waypoint is opened and stash not found because of that, check for that
+            if self._template_finder.search("WAYPOINT_MENU", self._screen.grab()).valid:
+                keyboard.send("esc")
+        start = time.time()
+        while time_out is None or (time.time() - start) < time_out:
+            template_match = self._template_finder.search(template_type, self._screen.grab(), threshold=threshold)
+            if template_match.valid:
+                x_m, y_m = self._screen.convert_screen_to_monitor(template_match.position)
+                if self._skill_hotkeys["telekinesis"] and telekinesis:
```

Comment:
Instead of a another nested if here you could check at the very beginning like this:
```python
if not (self._skill_hotkeys["telekinesis"] and telekinesis):
  return super().select_by_template(tempalte_type, success_func, time_out, threshold)
# Implement telekinsis specific implementation
...
```